client/src/pages/JobApplication/JobApplication.js

```diff
@@ -0,0 +1,213 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { useHistory } from 'react-router';
+import Alert from '@mui/material/Alert';
+import axios from '../../backendConnection';
+import './JobApplication.css';
+
+const JobApplication = ({ saveUserToken }) => {
+  const [formValues, setFormValues] = useState({
+    username: '',
+    email: '',
+    password: '',
+    password2: '',
+  });
+
+  // Keeps track of error message to display to user when the submit the form
+  const [error, setError] = useState('');
+
+  let history = useHistory();
+
+  const handleFormUpdate = (param) => (event) => {
+    setFormValues({ ...formValues, [param]: event.target.value });
+
+    // Remove error message when user is typing
+    if (error) setError('');
+  };
+
+  // submit form
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    if (
+      !formValues.username ||
+      !formValues.email ||
+      !formValues.password ||
+      !formValues.password2
+    ) {
+      setError('All field are required!');
+      return;
+    }
+
+    if (formValues.password !== formValues.password2) {
+      setError('Passords are not matching');
+      return;
+    }
+
+    try {
+      const response = await axios.post('/auth/signup', {
+        email: formValues.email,
+        username: formValues.username,
+        password: formValues.password,
+        password2: formValues.password2,
+      });
+
+      // Save user token
+      saveUserToken(response.data.token);
+      history.push('/');
+    } catch (error) {
+      setError(error.message);
+    }
```

Comment:
You would need to change this to match the form; this was for signup I believe

Reply:
anything else to change?